llama-index-core/pyproject.toml

```diff
@@ -71,6 +71,7 @@ pillow = ">=9.0.0"
 PyYAML = ">=6.0.1"
 wrapt = "*"
 pydantic = ">=2.7.0,<3.0.0"
+filetype = "^1.2.0"  # Used for multi-modal MIME utils
```

Comment:
I know theres's several libraries that offer this functionality in python -- curious what the rational was for landing on this one? (Just trying to be thorough for core dependencies)

Reply:
I'm open to alternatives.  That one came from the [existing Anthropic utils implementation](https://github.com/run-llama/llama_index/blob/5399050a5735fd72b35471f178e0c95d6d2aef43/llama-index-integrations/multi_modal_llms/llama-index-multi-modal-llms-anthropic/llama_index/multi_modal_llms/anthropic/utils.py#L4).  

I did some research into whether that should remain, but `python-magic` has a C dependency.  `filetype` has a [StackOverflow endorsement](https://stackoverflow.com/questions/10937350/how-to-check-type-of-files-without-extensions) as an alternative, which doesn't carry additional dependencies.